### ISSUE TITLE
Maven module / project layout restructuring - Notification and Persistence

### ISSUE THREAD

**Issue by prb112** [IBM/FHIR#1232]
A follow on to #1136

things to do
- move FHIRNotificationSubscriber, FHIRPersistence, and FHIRPersistenceInterceptor to fhir-server module
- move fhir-persistence-X projects into subfolder

**Comment by lmsurpre** [comment]
I started on the FHIRNotification refactor as part of #1136 but backed it out when I realized the scope of the edits that will be needed.  First we'd need to break the current dependency that fhir-server currently has on fhir-notification-websockets, fhir-notification-kafka, and fhir-notification-nats.  I was thinking to make these use the service-loader pattern like all our other extension points until I noticed that fhir-notification-websockets behaves much different from the others.
For the websockets one, fhir-server defines a websocket endpoint and instantiates a new FHIRNotificationSubscriber on each websocket connection.  So, for this one, I think we'd need to either:
A. move the websocket implementation into fhir-server; or
B. remove it altogether (given the issues mentioned at #1214 )
